tests/conftest.py

```diff
@@ -24,21 +26,31 @@
 from xclim.testing.utils import _default_cache_dir  # noqa
 from xclim.testing.utils import open_dataset as _open_dataset
 
-if not __xclim_version__.endswith("-dev") and helpers.TESTDATA_BRANCH == "main":
-    # This is fine on GitHub Workflows and ReadTheDocs
+if not __xclim_version__.endswith("-beta") and helpers.TESTDATA_BRANCH == "main":
+    # This does not need to be emitted on GitHub Workflows and ReadTheDocs
     if not os.getenv("CI") and not os.getenv("READTHEDOCS"):
         warnings.warn(
             f'`xclim` {__xclim_version__} is running tests against the "main" branch of `Ouranosinc/xclim-testdata`. '
-            "It is possible that changes in xclim-testdata may be incompatible with tests in this version. "
+            "It is possible that changes in xclim-testdata may be incompatible with test assertions in this version. "
             "Please be sure to check https://github.com/Ouranosinc/xclim-testdata for more information.",
             UserWarning,
         )
 
 if re.match(r"^v\d+\.\d+\.\d+", helpers.TESTDATA_BRANCH):
-    if parse_version(helpers.TESTDATA_BRANCH) > parse_version(__xclim_version__):
+    # Find the date of last modification of xclim source files to generate a calendar version
+    install_date = dt.strptime(
+        time.ctime(os.path.getmtime(working_set.by_key["xclim"].location)),
+        "%a %b %d %H:%M:%S %Y",
+    )
+    install_calendar_version = (
+        f"{install_date.year}.{install_date.month}.{install_date.day}"
```

Comment:
this regex will work for format `2023.05.12-r2` for example?

Reply:
`pkg_resources.parse_version` can handle modifiers in the version string.

For this particular line, we're creating a CalVer string from the date of last modification of xclim, therefore there won't be any modifiers.